src/main/python/fine_tuning/x_fold_cv.py

```diff
@@ -0,0 +1,139 @@
+"""
+Anserini: A toolkit for reproducible information retrieval research built on Lucene
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from __future__ import print_function
+import os
+import re
+import argparse
+import logging
+
+
+class XFlodValidate(object):
+    """
+    Perform X-Fold cross validation for various 
+    parameters and report the average effectiveness
+    for each fold
+    """
+    def __init__(self,output_root,fold=2):
+        self.logger = logging.getLogger('x_fold_cv.XFlodValidate')
+        self.output_root = output_root
+        self.eval_files_root = 'eval_files'
+        self.fold = fold
+
+    def _get_param_average(self):
+        # For each parameter set, get its 
+        # average performances in each fold,
+        # metric, reranking model, and base 
+        # ranking model
+        avg_performances = {}
+        for collection_name in os.walk(self.output_root).next()[1]:
```

Comment:
Using `.next()[1]` is very confusing...
Change it to sth like:
```
for sub_f in os.listdir(self.output_root):
   if os.is_file(sub_f):
      continue
   # if it is a folder blalala... 
```

The same for the `os.walk` below